py/src/braintrust/framework.py

```diff
@@ -207,14 +207,22 @@ class Evaluator:
     Whether to update an existing experiment with `experiment_name` if one exists. Defaults to false.
     """
 
+    timeout: Optional[float] = None
+    """
+    The duration, in seconds, after which to time out the evaluation.
+    Defaults to None, in which case there is no timeout.
+    """
+
 
 @dataclasses.dataclass
 class EvalResultWithSummary(SerializableDataClass):
     summary: ExperimentSummary
     results: List[EvalResult]
+    error: Optional[Exception] = None
+    exc_info: Optional[str] = None
```

Comment:
we're actually planning to do this per span (rather than at the top-level for an experiment). Is this used just to capture the timeout?

Reply:
Yup, just used to capture the timeout for now. Should we move this up to be at the span level right now?

Reply:
Propagated TimeoutError, per offline discussion